### PR TITLE
chore: ARM-318 - Tests for toast in module-new

### PR DESCRIPTION
**Pull request by jlews**
## What's new?

- Tests for toast in module-new

## Ticket number(s) in JIRA (if internal)

ARM-318

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [x] are your changes in Storybook?
- [x] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [x] does _everything_ have jsdoc?
- [x] is everything exported from index.ts?
- [x] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [x] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [x] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module-new/src/components/toast/toast.stories.tsx [hunk 1]
```diff
@@ -15,28 +16,76 @@ export default {
...
+    title: 'Here is a toast!',
+    description: 'Here is the description for my toast',
+  },
   decorators: [
     Story => (
       <ArmstrongProvider>
         <Story />
       </ArmstrongProvider>
     ),
   ],
```
[the remark this answers is not in the document]
Comment (judithrocketmakers) [reply module-new/src/components/toast/toast.stories.tsx:30]: E.g. https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators

File: module-new/src/components/toast/toast.stories.tsx [hunk 2]
```diff
@@ -15,28 +16,76 @@ export default {
...
 };
 
 /** stories */
 
 export const Default: StoryObj<typeof Toast> = {
   ...Template,
+  play: async ({ canvasElement, args }) => {
+    // launch toast
+    const button = within(canvasElement).getByText('Send a toast');
+    expect(button).toBeVisible();
```
Comment (judithrocketmakers) [inline module-new/src/components/toast/toast.stories.tsx:45]: As we're not testing the button itself we probably don't need this.

File: module-new/src/components/toast/toast.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (judithrocketmakers) [reply module-new/src/components/toast/toast.stories.tsx:30]: Ah, I see we're doing different provider configs each time so no, in this specific case it doesn't!

File: module-new/src/components/toast/toast.stories.tsx [hunk 3]
```diff
@@ -15,28 +16,76 @@ export default {
...
+  },
+  play: async ({ canvasElement, args }) => {
+    const button = within(canvasElement).getByText('Send a toast');
+    expect(button).toBeVisible();
+    userEvent.click(button);
+
+    const title = await findByText(document.body, args.title ?? '');
+    expect(title).toBeVisible();
+    // check auto-dismiss after custom dismiss time of 1000ms + 500ms for animation
+    await waitFor(() => expect(title).not.toBeVisible(), { timeout: 1500 });
```
Comment (judithrocketmakers) [inline module-new/src/components/toast/toast.stories.tsx:87]: I would be surprised if testing library didn't include mock timers - that would speed up the test and make it more reliable. Have a look at the docs and see if that's an option?

File: module-new/src/components/toast/toast.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlews) [reply module-new/src/components/toast/toast.stories.tsx:30]: I considered that, but I need to pass props to it as part of the toast tests, and it also seemed sensible to keep specific component providers contextual in storybook.

File: module-new/src/components/toast/toast.stories.tsx (hunk 3 above)
Comment (judithrocketmakers) [reply module-new/src/components/toast/toast.stories.tsx]: Urgh, yes, having just done a quick read-through it doesn't look like timer mocking is a thing in SB - which makes some sense when I think about it but still urgh.

Think your approach is the sensible one.

**Review by judithrocketmakers — APPROVED** [review]
Looking good - ready to merge.

**Pull request merged**